pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -987,6 +1017,16 @@ public void setObject(int parameterIndex, Object x) throws SQLException {
       setPGobject(parameterIndex, (PGobject) x);
     } else if (x instanceof Character) {
       setString(parameterIndex, ((Character) x).toString());
+      //#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.2"
+    } else if (x instanceof LocalDate) {
+      setDate(parameterIndex, java.sql.Date.valueOf((LocalDate) x));
+    } else if (x instanceof LocalTime) {
+      setTime(parameterIndex, Time.valueOf((LocalTime) x));
+    } else if (x instanceof LocalDateTime) {
+      setTimestamp(parameterIndex, Timestamp.valueOf((LocalDateTime) x));
+    } else if (x instanceof OffsetDateTime) {
+      setTimestamp(parameterIndex, Timestamp.from(((OffsetDateTime) x).toInstant()));
```

Comment:
Same here. As far as I understand, `OffsetDateTime` includes offset information, thus it could make sense to use `setTimestamp(..., Calendar)` API.